projects/go-lib/src/lib/components/go-select/go-select.component.ts

```diff
@@ -61,4 +189,30 @@ export class GoSelectComponent extends GoFormBaseComponent implements OnInit {
   onScroll($event: { start: number; end: number }): void {
     this.scroll.emit($event);
   }
+
+  onClose(): void {
+    this.emptyRefinedItems();
+  }
+
+  // store previous selected items incase of multiple and typeahead.
+  handleItemAdd(item: any): void {
+
+    if (!this.multiple || !this.typeahead) {
+      return;
+    }
+
+    this.previousSelectedItems.push(item);
+  }
+
+  // remove item from previous selected items incase of multiple and typeahead.
+  handleItemRemove(item: any): void {
+
+    if (!this.multiple || !this.typeahead ) {
+      return;
+    }
+
+    const index: number = this.previousSelectedItems.findIndex((prev: any) => prev[this.bindValue] === item.value[this.bindValue]);
+    this.previousSelectedItems.splice(index, 1);
+  }
+
```

Comment:
Just to make note, I see extra new lines throughout this whole file. Please remove in all functions.